### ISSUE TITLE
Get-DbaDatabase not failing on non existing database

### ISSUE THREAD

**Issue by andreasjordan** [sqlcollaborative/dbatools#7819]
### Verified issue does not already exist?

Yes

### What error did you receive?

No error - that is the problem.
Because Get-DbaDatabase is used by other commands like Set-DbaDbOwner, those commands also raise no error.

Is this the intended behaviour?

If not, the fix would not be so easy - we would have to completly rewrite this part:
https://github.com/sqlcollaborative/dbatools/blob/c8b825a5fc65031b7635202e77274765b43b2416/functions/Get-DbaDatabase.ps1#L284-L289

### Steps to Reproduce

```powershell
Get-DbaDatabase -SqlInstance srv1 -Database NonExistingDatabase -EnableException
Set-DbaDbOwner -SqlInstance srv1 -Database NonExistingDatabase -TargetLogin sa -EnableException
```

### Are you running the latest release?

Yes

### What PowerShell host was used when producing this error

Windows PowerShell (powershell.exe)

### PowerShell Host Version

does not matter

### SQL Server Edition and Build number

does not matter

**Comment by niphlod** [comment]
"give me a db with that name" --> not there --> not an error.
"set the owner of the db with that name" --> not there --> not an error.

IMHO the user needing to error out when the input data is not consistent with the "desired target" should just check if things exists beforehand.

**Comment by niphlod** [comment]
as to "how to behave on Set-" commands, it's another story, usually they do error out (by "PS standards") when the "thing" you want to "set" doesn't exist.

**Comment by niphlod** [comment]
Adding instead of editing: on the "let's error out if what you want to set didn't go through" side, this is also bad design https://github.com/sqlcollaborative/dbatools/blob/90474c562c7696c8957c38e8becbb4ca408f53a8/functions/Set-DbaDbOwner.ps1#L106-L109 . Continuing that way, you target a db that exists, but the set can't go through with it. 
The correct way is https://github.com/sqlcollaborative/dbatools/blob/90474c562c7696c8957c38e8becbb4ca408f53a8/functions/Set-DbaDbOwner.ps1#L121 , which when -EnableException is passed raises the exception.

tl;dr: lots of cmdlets skip over things and IMHO on user's side the "warn me if this didn't go through" should be handled as in:
- get-something
- does it exists ?
- yes, do something
- no, warn me
or
- do something
- did it return an object (optionally with status = succeeded or something like it) ?
- yes, fine
- no, warn me

dunno if we can even fathom erroring out on all Set-* functions TBH, we span a lot of functionality where an "harder-by-design" behaviour could be detrimental.

In any case, problem doesn't lie on Get-DbaDatabase, but on the functions using it.

**Comment by wsmelton** [comment]
It is by design. If you were to change it would require backward support because that would break a number of scripts in the wild.

This is the pattern we use on all of our Get functions, and is also the pattern used in PowerShell coding. Get functions should not output anything if the filter of the object is used to find it. The warning we do in `Get-DbaDatabase` was put there because it made users more comfortable since it is the number one command used. The only other Get commands that we spit warnings out are if they depend on some other feature existing (e.g. AG being enabled).

![image](https://user-images.githubusercontent.com/11204251/135695991-8f1d742c-01e4-4bc1-a54d-44f4de3bc693.png)

**Comment by andreasjordan** [comment]
Thanks for the discussion, that helped me a lot.
I will close this now, as there is no code change needed.
